src/cellfinder_core/detect/filters/volume/structure_detection.py

```diff
@@ -272,12 +257,12 @@ def merge_structures(
                 self.coords_maps.pop(neighbour_id)
                 self.obsolete_ids[neighbour_id] = updated_id
 
-    def structures_to_cells(self) -> List[Point]:
-        cell_centres = []
-        for iterator_pair in self.coords_maps:
-            structure = iterator_pair.second
+    def structures_to_cells(self) -> np.ndarray:
+        cell_centres = np.empty((len(self.coords_maps.keys()),3))
+        idx = 0
+        for structure_id, structure in self.coords_maps.items():
             p = get_structure_centre(structure)
-            cell_centres.append(p)
+            cell_centres[idx] = p
```

Comment:
Is this missing an increment of `idx`? It looks like the only index written to (and repeatedly overitten) at the moment is 0.

Reply:
😱 good catch!